api/src/Controller/Message/Get.php

```diff
@@ -93,4 +94,57 @@ public function preview(string $id): Response
 
         return new JsonResponse($message_preview, JsonResponse::HTTP_OK);
     }
+
+    /**
+     * @OA\Response(
+     *  response=200,
+     *  description="Get an array of message IDs from groups for the user's feed",
+     *  @OA\JsonContent(type="array", @OA\Items(type="integer"))
+     * )
+     *
+     * @OA\Tag(name="message")
+     *
+     * @Security(name="api_key")
+     */
+    #[Route('/feed', methods: ['GET'])]
+    public function feed(Request $request): Response
+    {
+        $this->denyAccessUnlessGranted('ROLE_USER');
+
+        $groupIds = $this->getUser()->getGroups()->map(fn($group) => $group->getId())->toArray();
+
+        $limit = $request->query->getInt('limit', 30);
```

Comment:
Nitpick: we maybe should put an upper bound on `$limit` (for example `100`)

Reply:
This is done in a new commit.